### ISSUE TITLE
a "BandROI" proposed

### ISSUE THREAD

**Issue by kklmn** [silx-kit/silx#3638]
Hello,

Herewith I 'd like to ask for a new ROI class that is a line ROI _with width_. It would look similar to ArcROI, just with three handles, not four, and with a straight shape. I suggest to optionally have "open" or "closed" state for both ends, when the roi continues infinitely ("open") or up to the end point ("closed"). I also suggest to add "polar1" and "polar2" modes of handle dragging when the roi is rotated about the corresponding end point.

Thank you in advance!

**Comment by t20100** [comment]
Hi,

There should be no problem to add such kind of ROI.
I'll give it a try for the coming release.

**Comment by kklmn** [comment]
Hi! Thanks for the implementation!

I've tried to test it in the current GH master, and get this error, something got lost:
```
Traceback (most recent call last):
  File "c:\ParSeq\parseq\tests\test_plotInteractiveImageROIwithKeys.py", line 10, in <module>
    from parseq.gui.roi import RoiWidgetWithKeyFrames
  File "c:\ParSeq\parseq\tests\../..\parseq\gui\roi.py", line 12, in <module>
    from silx.gui.plot.tools.roi import (
  File "C:\WPy64-31040\python-3.10.4.amd64\lib\site-packages\silx\gui\plot\__init__.py", line 62, in <module>
    from .PlotWidget import PlotWidget  # noqa
  File "C:\WPy64-31040\python-3.10.4.amd64\lib\site-packages\silx\gui\plot\PlotWidget.py", line 57, in <module>
    from ..colors import Colormap
  File "C:\WPy64-31040\python-3.10.4.amd64\lib\site-packages\silx\gui\colors.py", line 36, in <module>
    from silx.math import colormap as _colormap
  File "C:\WPy64-31040\python-3.10.4.amd64\lib\site-packages\silx\math\__init__.py", line 36, in <module>
    from .histogram import Histogramnd  # noqa
  File "C:\WPy64-31040\python-3.10.4.amd64\lib\site-packages\silx\math\histogram.py", line 145, in <module>
    from .chistogramnd import chistogramnd as _chistogramnd  # noqa
ModuleNotFoundError: No module named 'silx.math.chistogramnd'
```

**Comment by kklmn** [comment]
Sorry, I messed up the installation. Got it back working.

**Comment by kklmn** [comment]
Hi @t20100,

I've tested it, and it is almost there! I am missing the interaction mode, in the form of 1) RoiModeSelectorAction and 2) popup menu commands in the plot. 

Because the BandROI is not defined by its center, rotating around the center is not really valuable; I would instead turn it around one end when I am pulling the other end. Probably somebody would want it around the center, so rotation mode should be selectable.

I've also tested 'bounded', and for my purpose, it is good as is. I can imagine, however, that it can be useful as semi-bounded, so this property should not be an attribute of the roi but rather of each endpoint.

Thanks again!
